Initial implementation of org.crac API

tjwatson opened the pull request:
Initial work for #25594.
This includes a new (no-ship) `crac-1.0`.

tjwatson committed:
Implement CRaC APIs on Liberty checkpoint

Initial, no-ship, crac-1.3 feature that implements
org.crac third-party APIs that allows applications
to participate in the checkpoint/restore process.

tjwatson committed:
Enable Liberty Spring Boot 3.0 support to work with CRaC

This enables springBoot-3.0 for InstantOn but only when
running with the Liberty beta.  To do this a new directive
is added type:=beta

This prevents springBoot-3.0 from being used with InstantOn
checkpoint when using GA image of Liberty.

tjwatson commented:
#build

tjwatson commented:
#build

tevans78 commented on dev/com.ibm.websphere.appserver.features/visibility/public/crac-1.0/resources/l10n/io.openliberty.crac-1.0.properties:
```diff
@@ -0,0 +1,17 @@
+#NLS_MESSAGEFORMAT_NONE
+#
+
+description=This feature enables the CRaC APIs. 
```
Is this another temporary feature that will get absorbed into the kernel before GA? If not temporary then the description should be more verbose ... spell out "Coordinated Restore at Checkpoint"

tjwatson replied:
I doubt it will be folded into the kernel as it is providing third-party API.  I can update the description.

tevans78 replied:
And the copyright date then ;-)

anjumfatima90 committed:
Wait for the default_host to get configured with the default port on restore

Run the restore hook inside SpringBootApplicationImpl.SpringBootConfigImpl.installVirtualHostOnRestore
very last to make sure that the default_host is configured and started listening on default port.
Install the virtual host bundle with app port only if default endpoint was not configured.

tevans78 commented on dev/com.ibm.ws.kernel.boot.core/src/com/ibm/ws/kernel/boot/LaunchArguments.java:
```diff
@@ -322,6 +322,8 @@ private static String commandLineToInternalPhaseName(String phaseName) {
                 return CheckpointPhase.AFTER_APP_START.toString();
             case "beforeappstart":
                 return CheckpointPhase.BEFORE_APP_START.toString();
+            case "apprequested":
```
Copyright date looks wrong on this file too? I'm not going to check them all ;-)

tjwatson commented:
#build

dmuelle commented on dev/com.ibm.websphere.appserver.features/visibility/public/crac-1.0/resources/l10n/io.openliberty.crac-1.0.properties:
```diff
@@ -0,0 +1,17 @@
+#NLS_MESSAGEFORMAT_NONE
+#
+
+description=This feature enables the Coordinated Restore at Checkpoint org.crac APIs.
```
per IBM style, put code entity before noun identifiers
```suggestion
description=This feature enables the org.crac Coordinated Restore at Checkpoint  APIs.
```

tjwatson replied:
I'll update this, but I question sending messages off to translation for noship features at this time.

posmith commented:
L2 message review not required

tjwatson commented:
#build

tjwatson commented:
#build

tjwatson commented:
#build

tjwatson commented:
#build

tjwatson commented:
#build

tjwatson commented:
#build

tjwatson committed:
Add IBM copyright to source headers

tjwatson commented:
#libby
